merkle/merkle_path_test.go

```diff
@@ -10,31 +11,43 @@ type bitLenTestData struct {
 	expected int
 }
 
-type calcPathTestData struct {
+type auditPathTestData struct {
 	treeSize     int64
 	leafIndex    int64
 	expectedPath []storage.NodeID
 }
 
+type consistencyProofTestData struct {
+	priorTreeSize int64
+	treeSize      int64
+	expectedProof []storage.NodeID
+}
+
 // Expected paths built by examination of the example 7 leaf tree in RFC 6962. When comparing
 // with the document remember that our storage node layers are always populated from the bottom up.
 var expectedPathSize7Index0 = []storage.NodeID{storage.NewNodeIDForTreeCoords(0, 1, 64), storage.NewNodeIDForTreeCoords(1, 1, 64), storage.NewNodeIDForTreeCoords(2, 1, 64)}
 var expectedPathSize7Index3 = []storage.NodeID{storage.NewNodeIDForTreeCoords(0, 2, 64), storage.NewNodeIDForTreeCoords(1, 0, 64), storage.NewNodeIDForTreeCoords(2, 1, 64)}
 var expectedPathSize7Index4 = []storage.NodeID{storage.NewNodeIDForTreeCoords(0, 5, 64), storage.NewNodeIDForTreeCoords(0, 6, 64), storage.NewNodeIDForTreeCoords(2, 0, 64)}
 var expectedPathSize7Index6 = []storage.NodeID{storage.NewNodeIDForTreeCoords(1, 2, 64), storage.NewNodeIDForTreeCoords(2, 0, 64)}
 
+// Expected consistency proofs built from the examples in RFC 6962. Again, in our implementation
+// node layers are filled from the bottom upwards.
+var expectedConsistencyProofFromSize6To7 = []storage.NodeID{storage.NewNodeIDForTreeCoords(1, 2, 64), storage.NewNodeIDForTreeCoords(0, 6, 64), storage.NewNodeIDForTreeCoords(2, 0, 64)}
+var expectedConsistencyProofFromSize3To7 = []storage.NodeID{storage.NewNodeIDForTreeCoords(0, 2, 64), storage.NewNodeIDForTreeCoords(0, 3, 64), storage.NewNodeIDForTreeCoords(1, 0, 64), storage.NewNodeIDForTreeCoords(2, 1, 64)}
+var expectedConsistencyProofFromSize4To7 = []storage.NodeID{storage.NewNodeIDForTreeCoords(2, 1, 64)}
+
```

Comment:
Definitely needs more tests. There's a TODO for that but this will unblock getting the other pieces in place.